flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/RawToStringCastRule.java

```diff
@@ -57,11 +94,19 @@ protected String generateCodeBlockInternal(
                 .ifStmt(
                         deserializedObjTerm + " != null",
                         thenWriter ->
-                                thenWriter.assignStmt(
-                                        returnVariable,
-                                        CastRuleUtils.staticCall(
-                                                BINARY_STRING_DATA_FROM_STRING(),
-                                                methodCall(deserializedObjTerm, "toString"))),
+                                CharacterFamilyTrimmingAndPaddingCastRule
```

Comment:
I think we can keep this rule unmodifed and let `CharacterFamilyTrimmingAndPaddingCastRule` fire.

Reply:
Not really, because this rule returns a code block and not a single expression, so our input is not already `BinaryStringData` as it happens for example with `BooleanToString`.